tests/dags/maintenance/test_check_silenced_alerts.py

```diff
@@ -0,0 +1,136 @@
+from unittest import mock
+
+import pytest
+from airflow.exceptions import AirflowException
+from maintenance.check_silenced_alerts.check_silenced_alerts import (
+    check_configuration,
+    get_dags_with_closed_issues,
+    get_issue_info,
+)
+
+from tests.factories.github import make_issue
+
+
+@pytest.mark.parametrize(
+    "silenced_dags, dags_to_reenable, should_send_alert",
+    (
+        # No Dags to reenable, don't alert
+        ({}, [], False),
+        # One DAG to reenable
+        (
+            {"dag_a_id": "https://github.com/WordPress/openverse/issues/1"},
+            [
+                ("dag_a_id", "https://github.com/WordPress/openverse/issues/1"),
+            ],
+            True,
+        ),
+        # Multiple DAGs to reenable
+        (
+            {
+                "dag_a_id": "https://github.com/WordPress/openverse/issues/1",
+                "dag_b_id": "https://github.com/WordPress/openverse/issues/2",
+            },
+            [
+                ("dag_a_id", "https://github.com/WordPress/openverse/issues/1"),
+                ("dag_b_id", "https://github.com/WordPress/openverse/issues/2"),
+            ],
+            True,
+        ),
+    ),
+)
+def test_check_configuration(silenced_dags, dags_to_reenable, should_send_alert):
+    with mock.patch(
+        "maintenance.check_silenced_alerts.check_silenced_alerts.Variable",
+        return_value=silenced_dags,
+    ), mock.patch(
+        "maintenance.check_silenced_alerts.check_silenced_alerts.get_dags_with_closed_issues",
+        return_value=dags_to_reenable,
+    ) as get_dags_with_closed_issues_mock, mock.patch(
+        "maintenance.check_silenced_alerts.check_silenced_alerts.send_alert"
+    ) as send_alert_mock:
```

Comment:
nit: If there's any way to reasonably split this into separate `mock.patch` calls into separate expressions without introducing the nesting that's avoided here, that would be appreciated.

One option, if you're okay with not using a context manager and having the mock last the lifetime of the test (which should be fine with the current implementation) is pytest's [`monkeypatch`](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html) fixture. It doesn't require a context manager so can be split into multiple lines without having to put them into a single expression.

Reply:
It's also worth noting that (thanks to python 3.10) this block can be wrapped in parenthesis and `black` will format it a bit better:
```python
    with (
        mock.patch(
            "maintenance.check_silenced_dags.check_silenced_dags.Variable",
            return_value=silenced_dags,
        ),
        mock.patch(
            "maintenance.check_silenced_dags.check_silenced_dags.get_dags_with_closed_issues",
            return_value=dags_to_reenable,
        ) as get_dags_with_closed_issues_mock,
        mock.patch(
            "maintenance.check_silenced_dags.check_silenced_dags.send_alert"
        ) as send_alert_mock,
    ):
```